examples/apps/color_finder.py

```diff
@@ -0,0 +1,740 @@
+    return to_pil(image_array.transpose(1, 2, 0).astype(numpy.uint8))
+
+# from_pil (fn PIL image => image array): converts PIL image to image array.
+
+def from_pil(image):
+    image_array = numpy.asarray(image)
+    image_array.flags.writeable = True
+    return image_array
+
+# to_pil (fn image array => PIL image): converts image array to PIL image.
+
+def to_pil(image_array):
+    return Image.fromarray(numpy.uint8(image_array))
+
+
+LOOK_AROUND_STATE = 'look_around'
+FOUND_COLOR_STATE = 'found_color'
+DRIVING_STATE = 'driving'
+
+CAMERA_WIDTH = 320
+CAMERA_HEIGHT = 240
+
+
+class ColorFinder(cozmo.annotate.Annotator):
+    '''Cozmo looks around and drives after colors.
+
+    Cozmo's camera view is approximated into a matrix of colors.
+    Cozmo will look around for self.color_to_find, and once it is spotted, 
+    he will drive in the direction of that color.
+
+    Args:
+        robot (cozmo.robot.Robot): instance of the robot connected from run_program.
+
+    '''
+    def __init__(self, robot: cozmo.robot.Robot):
+
+        self.robot = robot
+        self.robot.camera.image_stream_enabled = True
+        self.robot.camera.color_image_enabled = True
+        self.fov_x  = self.robot.camera.config.fov_x
+        self.fov_y = self.robot.camera.config.fov_y
+        self.robot.add_event_handler(cozmo.objects.EvtObjectTapped, self.on_cube_tap)
+        self.robot.add_event_handler(cozmo.world.EvtNewCameraImage, self.on_new_camera_image)
+
+        self.color_selector_cube = None # type: LightCube
+        self.color_to_find = 'blue'
+        self.possible_colors_to_find = ['green', 'yellow', 'blue', 'red']
+        self.color_to_find_index = self.possible_colors_to_find.index(self.color_to_find)
+
+        self.grid_cube = None # type: LightCube
+        self.robot.world.image_annotator.add_annotator('color_finder', self)
+        self.robot.world.image_annotator.annotation_enabled = False
+        self.enabled = True
+        self.downsize_width = 32
+        self.downsize_height = 24
+        self.pixel_matrix = MyMatrix('white', self.downsize_width, self.downsize_height)
+
+        self.amount_turned_recently = radians(0)
+        self.moving_threshold = radians(9)
+
+        self.state = LOOK_AROUND_STATE
+
+        self.look_around_behavior = None # type: LookAroundInPlace behavior
+        self.drive_action = None # type: DriveStraight action
+        self.tilt_head_action = None # type: SetHeadAngle action
+        self.rotate_action = None # type: TurnInPlace action
+        self.lift_action = None # type: SetLiftHeight action
+
+    def apply(self, image, scale):
+        '''Draws a pixelated grid of Cozmo's approximate camera view
+        onto the viewer window.
+            
+        WM and HM are multipliers that scale the dimensions of the grid squares 
+        based on self.downsize_width and self.downsize_height
+
+        '''
+        d = ImageDraw.Draw(image)
+        WM = 20*32/self.downsize_width
```

Comment:
similarly to camerawidth and cameraheight, i would prefer if magic numbers like 20,24,32 were either global constants, or __init__ params fed into the class such that its resolution could be easily tuned